extensions/amp-list/0.1/test/test-amp-list.js

```diff
@@ -176,137 +172,103 @@ describes.realWin('amp-list component', {
-    templatesMock.expects('findAndRenderTemplateArray').withArgs()
-        .returns(renderPromise).once();
-    return list.layoutCallback().then(() => {
-      return Promise.all([fetchPromise, renderPromise]);
-    }).then(() => {
-      expect(fakeBind.rescanAndEvaluate).to.have.been.calledOnce;
-      expect(fakeBind.rescanAndEvaluate).calledWithExactly(rendered);
+    const output = [doc.createElement('div')];
+    const rendered = expectFetchAndRender(items, output);
+
+    return list.layoutCallback().then(() => rendered).then(() => {
+      expect(bind.scanAndApply).to.have.been.calledOnce;
+      expect(bind.scanAndApply).calledWithExactly(output, [list.container_]);
     });
   });
 
-  it('should reload data if the src attribute changes', () => {
-    const initialItems = [
-      {title: 'Title1'},
-    ];
-    const newItems = [
-      {title: 'Title2'}, {title: 'Title3'},
-    ];
-    const itemElement = doc.createElement('div');
-    const itemElement2 = doc.createElement('div');
-    const itemElement3 = doc.createElement('div');
-    const fetchPromise = Promise.resolve(initialItems);
-    const renderPromise = Promise.resolve([itemElement]);
-    listMock.expects('fetch_').withExactArgs('items')
-        .returns(fetchPromise).once();
-    templatesMock.expects('findAndRenderTemplateArray').withExactArgs(
-        element, initialItems)
-        .returns(renderPromise);
-    listMock.expects('getVsync').returns({
-      measure: () => {},
-    }).twice();
-    return list.layoutCallback().then(() => {
-      return Promise.all([fetchPromise, renderPromise]);
-    }).then(() => {
-      expect(list.container_.contains(itemElement)).to.be.true;
-      const newFetchPromise = Promise.resolve(newItems);
-      const newRenderPromise = Promise.resolve([itemElement2, itemElement3]);
-      listMock.expects('fetch_').withExactArgs('items')
-          .returns(newFetchPromise).once();
-      templatesMock.expects('findAndRenderTemplateArray').withExactArgs(
-          element, newItems)
-          .returns(newRenderPromise).once();
-      const spy = sandbox.spy(list, 'fetchList_');
-      element.setAttribute('src', 'https://data2.com/list.json');
-      list.mutatedAttributesCallback({'src': 'https://data2.com/list.json'});
-      expect(spy).to.be.calledOnce;
+  it('should _not_ reload if [src] attribute changes (before layout)', () => {
+    const items = [{title: 'foo'}];
+    const foo = doc.createElement('div');
+    const rendered = expectFetchAndRender(items, [foo]);
+
+    return list.layoutCallback().then(() => rendered).then(() => {
+      expect(list.container_.contains(foo)).to.be.true;
+
+      // Not allowed before layout.
+      listMock.expects('fetchList_').never();
+
+      element.setAttribute('src', 'https://new.com/list.json');
+      list.mutatedAttributesCallback({'src': 'https://new.com/list.json'});
+    });
+  });
+
+  it('should reload if [src] attribute changes (after layout)', () => {
```

Comment:
How do you explicitly make the test lay out or not lay out the element? I'm confused about the difference in setup between this and the previous test

Reply:
It's based on the value of `getLayoutWidth()`. Added another expectation and comments to help clarify.

Reply:
Actually simplified this test even more. Thanks for pointing this out.